util/uniq/slice.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package uniq provides dedulication utilities.
```

Comment:
typo: deduplication

But I worry about reading call sites of this code and reading the name `uniq.Slice(...)` and thinking it returns a unique (non-aliasing?) slice, rather than a de-duping a slice in place. I guess the call site will always have an `&` sign at least.

Did you consider `dedup` as a package name? What else are you planning on adding here?

Always things that modify in-place or return copies? If a mix, maybe Slice needs to be clear that it's in-place?

Reply:
I was thinking of this as an in-place analog to unix `uniq`, particularly since the call sites will often look like `sort.Slice; sort.Uniq`. But maybe that's too cute.

I didn't have plans to add anything else to this package.

Yeah, I was hoping the `&` at the call site would make the modify-in-place nature clear.

Happy to switch to dedup if you think it's better. After discussion, WDYT?

Reply:
I'd still go for explicit. `dedup.ModifySlice(&s.ids)` might be redundantly explicit (the package name already sounds like a verb). So either `dedup.Slice(&s)` or `uniq.ModifySlice(&s)` SGTM. Your call. I just think some part needs to suggest it modifies it in place a but more explicitly than the `&`.

Reply:
On further reflection, I'm worried that dedup sounds like it might do a slice-wide deduplication, not just adjacent elements.

`uniq.InPlace`? `dedup.Adjacent`? 

`man uniq` says

> uniq -- report or filter out repeated lines in a file

(Also, I had forgotten that `uniq` had so many options.)

Unless I hear otherwise, I guess I'll stick with uniq.ModifySlice.

Reply:
Now `uniq.ModifySlice`. Some docs updated as well.